packages/visual-editing/src/ui/useChannel.tsx

```diff
@@ -12,19 +12,19 @@ import {
  * @internal
  */
 export function useChannel(): ChannelsNode<Sends, Receives> | undefined {
-  const channelRef = useRef<ChannelsNode<Sends, Receives>>()
+  const [channel, setChannel] = useState<ChannelsNode<Sends, Receives>>()
 
   useEffect(() => {
-    const channel = createChannelsNode<Sends, Receives>({
+    const channelInstance = createChannelsNode<Sends, Receives>({
       id: 'overlays' satisfies VisualEditingConnectionIds,
       connectTo: 'presentation' satisfies VisualEditingConnectionIds,
     })
-    channelRef.current = channel
+    setChannel(channelInstance)
     return () => {
-      channel.destroy()
-      channelRef.current = undefined
+      channelInstance.destroy()
+      setChannel(undefined)
```

Comment:
I don't think it's necessary to call `setChannel(undefined)` here
```suggestion
```
as the `useChannel` hook is used during render, and React isn't going to call teardown of this useEffect and then call another render.

It's always going to be either:
```
render <VisualEditing /> -> calls useChannel()
run useChannel -> useEffect setup
// if strict mode
run useChannel -> useEffect teardown
run useChannel -> useEffect setup
render <VisualEditing /> -> calls useChannel()
run useChannel -> useEffect, deps is [] and so it's skipped
```
Or, in the event of unmounting (`enableVisualEditing` calls `root.unmount()`):
```
// Initial render (not strict mode)
render <VisualEditing /> -> calls useChannel()
run useChannel -> useEffect setup
// root.unmount()
run useChannel -> useEffect teardown
// instead of doing a last render of <VisualEditing />, its related DOM nodes are removed and `useRef` state is freed up
```

Reply:
Sorry @stipsan, just saw I made the change with no response to your comment (I feel I didn't hit submit), hope that didn't seem passive aggressive. 😆